docs/DMS/PRIMARY-DATA-STORAGE/RELATIONAL-QUERIES.md

```diff
@@ -44,36 +48,68 @@ erDiagram
     }
 ```
 
-For example, `QueryStudent` has a foreign key to the `Documents` table with a row per Student document. Because
-the query tables include the `Documents` partition key, query tables act as a cross-partition index on the
-Documents table while avoiding the downsides of an actual cross-partition index. For this reason, "Get All"
-queries will use these tables as well.
+### Architecture Details
+
+#### Foreign Key Relationships
+Each query table (e.g., `QueryStudent`) maintains a foreign key to the `Documents` table with one row per document. By including the `Documents` partition key, query tables effectively act as cross-partition indexes while avoiding the performance penalties of actual cross-partition indexes.
+
+#### Queryable Fields
+The columns in each query table correspond to fields that are available to API users for GET-by-query operations. This column list is derived from the `queryFieldMapping` in `ApiSchema.json`, which provides:
+- The query field name
+- The JSONPath to the document element location
+
+Note that queryable document fields are always scalar values, so a JSONPath will never return multiple values.
+
+#### Query Operations
+- **Pagination**: Operates exclusively on query tables
+- **"Get All" queries**: Use query tables for efficient cross-partition access
+- **Indexing strategy**: Individual query fields will be indexed on a case-by-case basis to balance insert performance with query performance
+
+> [!NOTE]
+> Deployments requiring fast ad hoc query performance should use DMS's search engine option instead.
```

Comment:
This really depends on the use case for "ad hoc query".

If your query is for one resource type, then the search DB option is _probably_ better (though not truly proven yet).

If your query needs to join rows, then the search engine option is not viable.